### PR TITLE
Introducing tfe_init and settings modules

### PR DESCRIPTION
**Pull request by jkerry**
## Background

This change consolidated the settings file management and cloud init script management with the other terraform enterprise cloud modules by consuming [the settings module](https://github.com/hashicorp/terraform-random-tfe-utility/tree/gcp_init_utilities/modules/settings) and [the tfe_init modules](https://github.com/hashicorp/terraform-random-tfe-utility/tree/gcp_init_utilities/modules/tfe_init).

This pull request should be paired with [this change](https://github.com/hashicorp/terraform-random-tfe-utility/pull/16)

## How Has This Been Tested

- [x] `/test all` has passed
- [x] nightly standalone rhel-8 worker test has passed in ptfe-replicated

## This PR makes me feel

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201752311630258

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -96,52 +96,99 @@ module "redis" {
...
+  tls_vers                 = var.tls_vers
+  metrics_endpoint_enabled = var.metrics_endpoint_enabled
+  custom_image_tag         = var.custom_image_tag
+  capacity_concurrency     = var.capacity_concurrency
+  capacity_memory          = var.capacity_memory
+
+  extra_no_proxy = concat([
+    local.common_fqdn,
+    var.networking_subnet_range],
+    local.rhel_no_proxy
```
Comment (anniehedgpeth) [inline main.tf:122]: Add if statement for RHEL

**Review by anniehedgpeth — APPROVED** [review]

**Review by sudomateo — APPROVED** [review]

**Pull request merged**